end_to_end/plugin_test.go

```diff
@@ -306,6 +308,42 @@ var _ = Describe("End to End plugin tests", func() {
 					Skip("This test is only needed for the most recent backup versions")
 				}
 			})
+			It("Will not hang if gpbackup and gprestore runs with single-data-file and plugin fails at init for restore", func(ctx SpecContext) {
+				pluginExecutablePath := fmt.Sprintf("%s/src/github.com/greenplum-db/gpbackup/plugins/example_plugin.bash", os.Getenv("GOPATH"))
+				copyPluginToAllHosts(backupConn, pluginExecutablePath)
+
+				testhelper.AssertQueryRuns(backupConn, "CREATE TABLE t0(a int);")
+				testhelper.AssertQueryRuns(backupConn, "INSERT INTO t0 SELECT i FROM generate_series(1, 10)i;")
+				defer testhelper.AssertQueryRuns(backupConn, "DROP TABLE t0;")
+
+				timestamp := gpbackup(gpbackupPath, backupHelperPath,
+					"--single-data-file",
+					"--plugin-config", pluginConfigPath)
+
+				backupCluster.GenerateAndExecuteCommand(
+					"Instruct plugin to fail",
+					cluster.ON_HOSTS,
+					func(contentID int) string {
+						return fmt.Sprintf("touch /tmp/GPBACKUP_PLUGIN_FAIL")
+					})
+
+				gprestoreCmd := exec.Command(gprestorePath,
+					"--timestamp", timestamp,
+					"--redirect-db", "restoredb",
+					"--plugin-config", pluginConfigPath)
+
+				_, err := gprestoreCmd.CombinedOutput()
+				Expect(err).To(HaveOccurred())
```

Comment:
Is this an expected error?
```
[ERROR]:-Expected to restore 10 rows to table t0, but restored 0 instead
```
Can't partial restore be achieved using the --on-error-continue option if an error occurs on one segment?

Reply:
> Is this an expected error?

yes, in the test plugin fails at start up when doing restore, so 0 rows are restored.

> Can't partial restore be achieved using the --on-error-continue option if an error occurs on one segment?

According to code it shouldn't be so - if `getRestoreDataReader` returns an error, the code in `doRestoreAgent` doesn't take `onErrorContinue` into account. It will just return as soon as possible.